qendpoint-store/src/main/java/com/the_qa_company/qendpoint/store/EndpointStoreQueryPreparer.java

```diff
@@ -128,15 +128,17 @@ protected CloseableIteration<? extends BindingSet, QueryEvaluationException> eva
 			new DisjunctiveConstraintOptimizer().optimize(tupleExpr, dataset, bindings);
 			new SameTermFilterOptimizer().optimize(tupleExpr, dataset, bindings);
 			new QueryModelNormalizerOptimizer().optimize(tupleExpr, dataset, bindings);
-			new QueryJoinOptimizer(evaluationStatistics).optimize(tupleExpr, dataset, bindings);
+			new QueryJoinOptimizer(evaluationStatistics, tripleSource).optimize(tupleExpr, dataset, bindings);
```

Comment:
Including the tripleSource here is what allows the QueryJoinOptimizer to check the possible statement orders from the underlying store.